modules/c++/xml.lite/unittests/test_Speed.cpp

```diff
@@ -0,0 +1,413 @@
+    "]]>"
+    "</foo>"
+    "</root>";
+
+  TEST_ASSERT_EQ(writexml(xml), truth);
+}
+//-----------------------------------------------------------------
+//-----------------------------------------------------------------
+TEST_CASE(normative_use_case)
+{
+  SETUP_XML;
+  xml.declareNamespace("","","http://zombo.com");
+  xml.setData("A>one", 1);
+  xml.setData("A>three", 3);
+  xml.setData("A>two", 2); // order matters (but XML doesn't care)
+  xml.declareNamespace("A", "num", "urn:numbers/counting");
+  xml.setNamespace("A", "num");
+  std::string bpoly("B>razzle>poly2d");
+  xml.addAttribute(bpoly, "order", 2);
+  xml.setData(bpoly+">x", 6.123);
+  xml.setNextData(bpoly+">x", 1.354);
+  xml.setData(bpoly+">y", 5.098);
+  xml.setNextData(bpoly+">y", 0.415);
+  xml.setData("B>razzle>name", "soopah razzle");
+
+
+  std::string truth = 
+    "<root xmlns=\"http://zombo.com\">"
+    "<num:A xmlns:num=\"urn:numbers/counting\">"
+    "<num:one>1</num:one><num:three>3</num:three><num:two>2</num:two></num:A>"
+    "<B><razzle>"
+    "<poly2d order=\"2\">"
+    "<x>6.123</x><x>1.354</x><y>5.098</y><y>0.415</y>"
+    "</poly2d>"
+    "<name>soopah razzle</name>"
+    "</razzle></B>"
+    "</root>";
+
+  TEST_ASSERT_EQ(writexml(xml), truth);
+}
+//-----------------------------------------------------------------
+TEST_CASE(create_from_existing)
+{
+  SETUP_XML;
+  xml.declareNamespace("","","http://zombo.com");
+  xml.setData("A>one", 1);
+  xml.setData("A>three", 3);
+  xml.setData("A>two", 2); 
+
+  xml::lite::Speed foo;
+  foo.fromDocument(&xml, true);
+  foo.setData("lorem>ipsum>dolor>sit>amet", "consectetur adipiscing elit");
+
+  std::string truth = 
+    "<root xmlns=\"http://zombo.com\">"
+    "<A><one>1</one><three>3</three><two>2</two></A>"
+    "<lorem><ipsum><dolor><sit><amet>"
+    "consectetur adipiscing elit"
+    "</amet></sit></dolor></ipsum></lorem>"
+    "</root>";
+
+  TEST_ASSERT_EQ(writexml(foo), truth);
+  // xml is still a view of the same document, it just doesn't own the root
+  // node anymore. This means that it should see all the updates done to foo.
+  TEST_ASSERT_EQ(writexml(xml), truth);
+}
+//=================================================================
+TEST_CASE(created_from_existing_no_stealing)
+{
+  SETUP_XML;
+  xml.declareNamespace("","","http://zombo.com");
+  xml.setData("A>one", 1);
+  xml.setData("A>three", 3);
+  xml.setData("A>two", 2); 
+
+  xml::lite::Speed foo;
+  foo.fromDocument(&xml, false);
+  foo.setData("lorem>ipsum>dolor>sit>amet", "consectetur adipiscing elit");
+
+  std::string truth = 
+    "<root xmlns=\"http://zombo.com\">"
+    "<A><one>1</one><three>3</three><two>2</two></A>"
+    "<lorem><ipsum><dolor><sit><amet>"
+    "consectetur adipiscing elit"
+    "</amet></sit></dolor></ipsum></lorem>"
+    "</root>";
+
+  TEST_ASSERT_EQ(writexml(foo), truth);
+}
+//=================================================================
+int main (int argc, char** argv)
```

Comment:
To avoid warnings about unused variables, either remove or comment out these names:
`int main(int, char**)`